### PR TITLE
Update TextLink support Theming

### LINKED ISSUE #197: TextLink href prop
`<TextLink>` has `url` props. I suggest to have `href`.
In Next.js you can generate routes between pages with `<Link>`. It passes url into `<a href=...>` in `render`
```jsx
<Link href="/bookings">
  <a>link</a>
</Link>
```
It could even pass `href` from `<Link>` to `<TextLink>` by `<Link passHref href="/booking">` but only when `<TextLink>` has `href` not `url`.

Second reason for `href` is that `<a href="mailto:xxx@xxx.com">` is not url, but it's still **H**ypertext **REF**erence.

----

Make sure these boxes are ticked before submitting a new issue:

* [x] I have checked that this is not a duplicate
* [x] I have added a label categorizing this under BLKN | Kiwicom | Icons | Global

### PR DESCRIPTION
**Pull request by williamkolmacka**
Added support for theming
Rewrited to styled components
Changed prop url to href
Updated tests
Added flow strict type
Added constants for type prop

Closes #197

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "build": "yarn clean && yarn build:icons && yarn build:lib && yarn build:flow && yarn build:module",
     "build:lib": "babel --out-dir lib --ignore *.stories.js,*.test.js src",
     "build:flow": "flow-copy-source -v -i '**/*.?(stories|test).js' src lib",
-    "build:module": "babel --presets react --plugins styled-jsx/babel,babel-plugin-styled-components,transform-class-properties --no-babelrc --out-dir es --ignore *.stories.js,*.test.js src",
+    "build:module": "babel --presets react --plugins styled-jsx/babel,babel-plugin-styled-components,transform-class-properties,transform-object-rest-spread --no-babelrc --out-dir es --ignore *.stories.js,*.test.js src",
```
Comment (jaroslav-kubicek) [inline package.json]: topic for another PR - create `.babelrc` for module, this is already too weird

File: src/TextLink/index.js [hunk 2]
```diff
@@ -1,63 +1,77 @@
...
+    | "search"
+    | "tag",
+  theme: typeof defaultTokens,
 };
 
+const StyledTextLink = styled(({ tokens, theme, type, url, ...props }) => (
+  <a {...props}>{props.children}</a>
+))`
+      color: ${props => props.tokens.colorTextLink[props.type]};
+      font-weight: ${props => props.theme.fontWeightLinks};
```
Comment (jaroslav-kubicek) [inline src/TextLink/index.js]: what about object destructuring? `props => props.theme.fontWeightLinks` => `({ theme }) => theme.fontWeightLinks`

Comment (wEprous) [reply src/TextLink/index.js]: I like it. But in some components, we need more props to destructuring.
`height: ${({ tokens, size }) => tokens.heightButton[size]};`

So what do you like more? cc @jaroslav-kubicek @tomashapl @williamkolmacka

Comment (williamkolmacka) [reply src/TextLink/index.js]: I like style that @jaroslav-kubicek mentioned. From my point of view, it's more readable and typing only props you want is much better.

Comment (wEprous) [reply src/TextLink/index.js]: OK, so we need to use it everywhere then.

File: src/TextLink/__tests__/__snapshots__/index.test.js.snap [hunk 3]
```diff
@@ -1,63 +1,3 @@
...
-        "underline",
-        "underline",
-        "#00a991",
-      ]
-    }
-    styleId="211793505"
-  />
-</a>
-`;
+exports[`TextLink should match snapshot 1`] = `null`;
```
Comment (wEprous) [inline src/TextLink/__tests__/__snapshots__/index.test.js.snap]: the test probably cannot find anything and snap it, can you check it @williamkolmacka ?

File: src/TextLink/index.js [hunk 4]
```diff
@@ -1,63 +1,84 @@
...
+      outline-width: 3px;
+    }        
+  `;
+
 const TextLink = (props: Props) => {
-  const { children, url, onClick, newTab, type, size } = props;
+  const { children, url, external, theme, rel, onClick } = props;
+  const tokens = {
+    colorTextLink: {
+      primary: theme.colorLinkPrimary,
```
Comment (wEprous) [inline src/TextLink/index.js]: Define constant for type

**Comment by tomashapl** [comment]
@wEprous plz rebase 🙏

**Pull request merged**